torch/quantization/fx/prepare.py

```diff
@@ -725,28 +726,40 @@ def make_input_output_share_observers(
     node: Node,
     model: torch.nn.Module,
     modules: Dict[str, torch.nn.Module],
-) -> None:
+) -> bool:
```

Comment:
nit: if we are changing this to return `bool`, can we change the name to `maybe_make_input_output_share_observers`

optional nit: some people consider it an anti-pattern to have mutations return success or failure, one could decompose this into a function which returns whether the mutation should happen (return bool) and a function which performs the mutation (return None)

Reply:
sure, will change the name.
For the other comment, I agree it feels a bit weird to return True/False here, but in this case it looks like the check is not trivial, do we have other ways that we can improve the structure?

Reply:
one option would be to throw an exception on failure (instead of returning False), and catch it in the caller